app/containers/App/index.js

```diff
@@ -71,13 +72,23 @@ class App extends React.PureComponent {
     this.toggleMenu = this.toggleMenu.bind(this);
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.logged) {
+      this.setState({ rightMenuOpen: false });
+    }
+  }
+
   toggleMenu() {
     this.setState({ rightMenuOpen: !this.state.rightMenuOpen });
   }
 
+  isRightMenuOpened() {
+    return this.props.logged && this.state.rightMenuOpen;
```

Comment:
zamiast sprawdzać tutaj czy user jest zalogowany to sprawdzałbym to w funkcji `toggleMenu` ponieważ teraz jeżeli wylogowany user zrobi toggleMenu po czym się zaloguje to menu mu się wysunie